requests_toolbelt/utils/dump.py

```diff
@@ -0,0 +1,166 @@
+"""This module provides functions for dumping information about responses."""
+import collections
+
+from requests import compat
+
+
+__all__ = ('dump_response',)
+_PrefixSettings = collections.namedtuple('PrefixSettings',
+                                         ['request', 'response'])
+HTTP_VERSIONS = {
+    9: b'0.9',
+    10: b'1.0',
+    11: b'1.1',
+}
+
+
+class PrefixSettings(_PrefixSettings):
+    def __init__(self, request, response):
+        request = _coerce_to_bytes(request)
+        response = _coerce_to_bytes(response)
+        super(PrefixSettings, self).__init__(request, response)
+
+
+def _format_header(name, value):
+    return (_coerce_to_bytes(name) + b': ' + _coerce_to_bytes(value) +
+            b'\r\n')
+
+
+def _dump_request_data(request, prefixes, bytearr):
+    prefix = prefixes.request
+    method = _coerce_to_bytes(request.method)
+    uri = compat.urlparse(request.url)
+
+    request_path = _coerce_to_bytes(uri.path)
+    if uri.query:
+        request_path + b'?' + _coerce_to_bytes(uri.query)
+
+    # <prefix><METHOD> <request-path> HTTP/1.1
+    bytearr.extend(prefix + method + b' ' + request_path + b' HTTP/1.1\r\n')
```

Comment:
So unless `stream=True` the best bet seems to be looking at `response.connection.proxy_manager` (`r.connection` is the `HTTPAdapter` that sent the request/generated the response). So what does the format look like for this?

FWIW, if I do:

```py
>>> r = requests.get('http://httpbin.org/get', proxies={'http': 'http://localhost:8088'})
>>> r.connection.proxy_manager
{'http://localhost:8088': <requests.packages.urllib3.poolmanager.ProxyManager object at 0x10ba94f10>}
```

(That's assuming mitmproxy running on port 8088)